server/src/main/java/org/elasticsearch/index/mapper/RootObjectMapper.java

```diff
@@ -79,6 +80,24 @@ public Mapper.Builder<?> parse(String name, Map<String, Object> node, ParserCont
 
     @Override
     public RootObjectMapper merge(Mapper mergeWith) {
-        return (RootObjectMapper) super.merge(mergeWith);
+        RootObjectMapper newMapper = (RootObjectMapper) super.merge(mergeWith);
+        if (mergeWith instanceof RootObjectMapper rootObjectMapper) {
+            rootObjectMapper.columnPositionResolver.updateMaxColumnPosition(this.columnPositionResolver.getMaxColumnPosition());
+            newMapper.columnPositionResolver = ColumnPositionResolver.resolve(rootObjectMapper.columnPositionResolver);
+        }
+        return newMapper;
+    }
+
+    @Override
+    public ObjectMapper mappingUpdate(Mapper mapper) {
+        RootObjectMapper mappingUpdate = (RootObjectMapper) super.mappingUpdate(mapper);
+        // it makes sense to re-build the columnPositionResolver with respect to inner mapper changes,
+        // but it is not used, so cleared instead.
+        mappingUpdate.columnPositionResolver = new ColumnPositionResolver<>();
+        return mappingUpdate;
```

Comment:
This comment doesn't really explain much. Why does it make sense to rebuild it?

Reply:
I wanted to set an invariant that `ColumnPositionResolver` contains all unpositioned mappers and maxColumnPosition of all its children. So, I thought when `mappingUpdate` generates an instance of `RootObjectMapper`, the `ColumnPositionResolver` field should also reflect on the inner mapping changes.

Anyways, I have made the change to remove maxColumnPosition out of `ColumnPositionResolver`, [2eb7f70](https://github.com/crate/crate/pull/12754/commits/2eb7f701542e8b8ec3eff5af83f77cbf166a3bfb)

Please let me know what you think. Thank you